internal/providercache/dir_modify.go

```diff
@@ -30,6 +46,28 @@ func (d *Dir) InstallPackage(ctx context.Context, meta getproviders.PackageMeta,
 	)
 
 	log.Printf("[TRACE] providercache.Dir.InstallPackage: installing %s v%s from %s", meta.Provider, meta.Version, meta.Location)
+
+	unlock, err := d.lock(ctx, meta.Provider, meta.Version)
```

Comment:
I'm not sure if I'm reading this wrong, but it seems like by the time we get here we should already be holding the lock from the call in `Dir.InstallPackage`, and so I'd expect this to deadlock. :thinking: 

Am I missing something here? :thinking: 

(Having the outer function handle all the locking does seem like a neat solution to the `errcheck` lint failure I see a few lines below this comment at the time of writing.)

Reply:
Might have ran into an issue with a rebase.